### PR TITLE
fix(stress_thread): `stop_test_on_failure` affect log parsing

### PR DESCRIPTION
**Pull request by fruch**
when `stop_test_on_failure=False` is used by nemesis code, it was affecting only the return value check to not raise critical events, but if any log line that was raising critical events would still be raised and stop the test in the middle.

this change propgates the `stop_test_on_failure` into the log parsing code to act inlined with the rest of the error handling logic

Ref: #6543

### Testing
- [x] - https://jenkins.scylladb.com/view/staging/job/scylla-staging/job/fruch/job/longevity-mv-si-4days-test/6

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by ShlomiBalalis — APPROVED** [review]

File: sdcm/stress_thread.py [hunk 1]
```diff
@@ -63,6 +64,8 @@ def run(self) -> None:
 
                     if pattern.search(line):
                         event.add_info(node=self.node, line=line, line_number=line_number).publish()
+                        if event.severity == Severity.CRITICAL and not self.stop_test_on_failure:
```
Comment (juliayakovlev) [inline sdcm/stress_thread.py:67]: Do we need same for scylla-bench?

Comment (fruch) [reply sdcm/stress_thread.py:67]: I'm not sure we have critical log events in scylla-bench.

I'll check tomorrow if we have any

Comment (juliayakovlev) [reply sdcm/stress_thread.py:67]: AFAIR we can get critical event when stress command fails

**Review by soyacz — DISMISSED** [review]
LGTM

File: sdcm/stress_thread.py (hunk 1 above)
Comment (fruch) [reply sdcm/stress_thread.py:67]: scylla-bench only has this log error:
```
ScyllaBenchLogEvent.add_subevent_type("ConsistencyError", severity=Severity.ERROR, regex=r"received only")
```
so no criticals, and no need for implementing there for now.

Comment (juliayakovlev) [reply sdcm/stress_thread.py:67]: See this run: https://argus.scylladb.com/test/3190b0e9-a489-4e19-b855-32dc837b5431/runs?additionalRuns[]=e9fb4ce4-bec9-4429-995e-fc13f77a27bb

Comment (fruch) [reply sdcm/stress_thread.py:67]: as I said it's not log event, and we might have an issue with s-b logic that we should look at on it's own issue/PR

**Comment by fruch** [comment]
Last run now proved this fix is working as expected

File: sdcm/stress_thread.py [hunk 2]
```diff
@@ -62,6 +63,9 @@ def run(self) -> None:
                         event.event_id = self.event_id
 
                     if pattern.search(line):
+                        if event.severity == Severity.CRITICAL and not self.stop_test_on_failure:
+                            event = event.clone()  # so we don't change the severity to other stress threads
```
Comment (fruch) [inline sdcm/stress_thread.py:67]: Initially I've change the level only after publishing, which wasn't good enough

**Review by soyacz — APPROVED** [review]
LGTM

**Pull request merged**